mesh/src/main/java/no/nordicsemi/android/mesh/transport/GenericBatteryStatus.java

```diff
@@ -40,9 +40,9 @@ void parseStatusParameters() {
         final ByteBuffer buffer = ByteBuffer.wrap(mParameters).order(ByteOrder.LITTLE_ENDIAN);
         mBatteryLevel = buffer.get();
```

Comment:
Once we get rid of the byte buffer we could get the battery status as follows may be `mParameters[0] & 0xFF`?

Reply:
I think that the (& 0xFF) is not needed because the value is between 0x00 and 0x64. It can only be 0xFF if the battery level is unknown following the Mesh specifications, if I am right. But yes, like you said I will get the value from `mParameters` ! 😄